components/layout_2020/flow/root.rs

```diff
@@ -359,3 +375,72 @@ impl FragmentTree {
         .unwrap_or_else(Rect::zero)
     }
 }
+
+/// https://drafts.csswg.org/css-backgrounds/#root-background
+#[derive(Clone)]
+pub(crate) struct CanvasBackground {
+    pub style: Option<Arc<ComputedValues>>,
+
+    /// The element whose style we have above
+    pub from_element: OpaqueNode,
+
+    pub root_element: OpaqueNode,
```

Comment:
A doc comment might be useful here explaining the distinction between this element and `from_element`.